xdsl/transforms/experimental/live_range.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass
+from xdsl.ir import Block, OpResult, SSAValue
+from xdsl.utils.exceptions import InvalidIRException
+
+
+@dataclass(init=False, frozen=True)
+class LiveRange:
+    """
+    Represents the live range of an SSA value as half-open interval using the
+    index of an Operation within its parent Block.
+    """
+
+    value: SSAValue
+    start: int
```

Comment:
Indices are really inefficient for dealing with linked lists, these should probably be ops instead

Reply:
At some point you need to compare the ranges for the LS algorithm, because right now you don't have any order relation on ops.

Reply:
What does inefficient mean?
I doubt this will be a bottleneck unless we operate over hundreds of operations per block.
I'd rather avoid prematurely optimizing this.

Moreover, even LLVM's API uses numbering (`VNInfo` class), this is standard compiler treatment and it is way simpler to reason and handle numeric intervals, especially if splitting comes into play later.

Reply:
Why wouldn't we have hundreds of operations per block? It doesn't seem like a high number to me. We don't cache indices on operations, so this is O(n^2) in xDSL, which it isn't in LLVM/MLIR